### ISSUE TITLE
Drop interface for Model resource example

### ISSUE THREAD

**Issue by colmben** [troyanskiy/rest-core#28]
Hi,

Thanks again for this library, I am using it for a reasonably large project that I am upgrading from Angular 1.5 -> Angular 5.

One thing I noticed about your example code is that for the Model examples, you have them implementing an interface. I think this is a duplication that isn't required - you would either have an interface, or a Model class, not both? You can see that doing it with both means you have to dupe the member declarations in both places. So for the User example, you can drop the IUser interface completely and then User becomes:

```
export class User extends ResourceModel  {

  readonly $resource = UserResource;

  id: number;
  userName: string;
  firstName: string;
  lastName: string;
  groupId: string;
  
  fullName: string; // generated from first name and last name
  
  constructor(data?: User) {
    super();
    if (data) {
      this.$setData(data);
    }
  }
  
  $setData(data: User): this {
    Object.assign(this, data);
    this.fullName = `${this.firstName} ${this.lastName}`;
    return this;
  }
  
  toJSON() {
    // here i'm using lodash lib pick method.
    return _.pick(this, ['id', 'firstName', 'lastName', 'groupId']);
  }

}
```

**Comment by troyanskiy** [comment]
Hello.
Thank you for using the library.
Yes, you can drop the Interface, but I prefer to have it.
The decision to create or not to create Interfaces is in Developers' hands.

**Comment by colmben** [comment]
That is true of course :) I love Todd Moto's thoughts on all things Angular and he actually has a little article on this topic...

https://toddmotto.com/classes-vs-interfaces-in-typescript

**Comment by troyanskiy** [comment]
Thanks for sharing the article!

**Comment by troyanskiy** [comment]
Hello @all
I did some refactoring and repository restructuring.
From now on new repository is located here https://github.com/troyanskiy/ngx-resource
Please create new issues in new repo.
I will close all the issues in that repo.

Also I have released 7.0.0-beta.5 version with `Observables` support. Please test it and create new issues in new repo if you will find some bugs.

To update to new version run
`npm i --save @ngx-resource/core@beta @ngx-resource/handler-ngx-http@beta`

### Breaking changes:
#### `ResourceGlobalConfig`
* removed `asPromise` and `asResourceResponse`
* added `returnAs` with enum type `ResourceActionReturnType`; Default is `ResourceActionReturnType.Observable`

#### `IResourceParams` and `IResourceAction`
* removed `asPromise` and `asResourceResponse`
* added `returnAs` with enum type `ResourceActionReturnType`; Default is `ResourceActionReturnType.Observable`

### Deprecation
* `IResourceMethod` -> `IResourceMethodPromise`
* `IResourceMethodStrict` -> `IResourceMethodPromiseStrict`
* `IResourceMethodFull` -> `IResourceMethodPromiseFull`
* `IResourceMethodStrictFull` -> `IResourceMethodPromiseStrictFull`
* `ResourceCRUD` -> `ResourceCRUDPromise`

### New
* `ResourceCRUDObservable` - same as ResourceCRUDPromise, but typed to Observable

I still have to update Docs and provide migration guide.

Thank you @all !!!
